src/main/kotlin/no/nav/amt/distribusjon/journalforing/dokdistfordeling/DokdistfordelingClient.kt

```diff
@@ -0,0 +1,70 @@
+package no.nav.amt.distribusjon.journalforing.dokdistfordeling
+
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.request.header
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.http.ContentType
+import io.ktor.http.HttpHeaders
+import io.ktor.http.contentType
+import io.ktor.http.isSuccess
+import no.nav.amt.distribusjon.Environment
+import no.nav.amt.distribusjon.application.plugins.objectMapper
+import no.nav.amt.distribusjon.auth.AzureAdTokenClient
+import org.slf4j.LoggerFactory
+import java.util.UUID
+
+class DokdistfordelingClient(
+    private val httpClient: HttpClient,
+    private val azureAdTokenClient: AzureAdTokenClient,
+    environment: Environment,
+) {
+    private val scope = environment.safScope
+    private val url = environment.dokdistfordelingUrl
+    private val navCallId = Environment.appName
+    private val log = LoggerFactory.getLogger(javaClass)
+
+    suspend fun distribuerJournalpost(journalpostId: String, tvingSentralPrint: Boolean = false): UUID {
+        val token = azureAdTokenClient.getMachineToMachineToken(scope)
+        val response = httpClient.post("$url/rest/v1/distribuerjournalpost") {
+            header(HttpHeaders.Authorization, token)
+            header("Nav-Callid", navCallId)
+            contentType(ContentType.Application.Json)
+            setBody(objectMapper.writeValueAsString(DistribuerJournalpostRequest(journalpostId, tvingSentralPrint)))
+        }
+
+        if (!response.status.isSuccess()) {
+            error("Distrubering av journalpost $journalpostId feilet: ${response.status}")
+        }
+
+        val bestillingsId = response.body<DistribuerJournalpostResponse>().bestillingsId
+
+        log.info("Distribuerte journalpost $journalpostId, bestillingsId=$bestillingsId")
+
+        return bestillingsId
+    }
+}
+
+data class DistribuerJournalpostRequest(
+    val journalpostId: String,
+    val tvingSentralPrint: Boolean = false,
+    val dokumentProdApp: String = Environment.appName,
+    val distribusjonstype: Distribusjonstype = Distribusjonstype.VEDTAK,
+    val distribusjonstidspunkt: Distribusjonstidspunkt = Distribusjonstidspunkt.UMIDDELBART,
```

Comment:
Når vi velger denne som default så må vi være klar over at bruker som har SDP potensielt vil få varsel når som helst på døgnet, og det pleier å være lite populært. Vi vil jo normalt ikke sende noe her til de som har SDP, men hvis de ikke har sjekket meldingen på navno kan det jo skje (hvis ikke vi skal tvinge sentral print). Det kan godt være sånn for nå, men er nok noe vi bør ta stilling til ifm brev til de som ikke er innom navno.

Reply:
Vi kan sikkert sette kjernetid som default, også gå opp det senere 👍